implementation/jobberknoll/api/shared/contracts/account-dto.ts

```diff
@@ -0,0 +1,36 @@
+import { z } from "@hono/zod-openapi";
+import { uuid } from "@jobberknoll/core/shared";
+
+export const AccountDto = z.object({
+  id: z.string().uuid().openapi({
+    description: "Account identifier as UUIDv7.",
+    examples: [uuid()],
+  }),
+  type: z.enum(["passenger", "driver", "admin", "inspector"]).openapi({
+    description: "Type of the account.",
+    examples: ["passenger"],
+  }),
+  fullName: z.string().openapi({
+    description: "Full name of the account owner.",
+    examples: ["John Smith"],
+  }),
+  email: z.string().email().openapi({
+    description: "Email address of the account owner.",
+    examples: ["john.smith@example.com"],
+  }),
+  phoneNumber: z.string().optional().openapi({
+    description: "Phone number of the passenger.",
+    examples: ["+48 123 456 789"],
+  }),
+}).openapi("AccountDto", {
+  description: "Account details.",
+  examples: [{
+    id: uuid(),
+    type: "passenger",
+    fullName: "John Smith",
+    email: "john.smith@example.com",
+    phoneNumber: "+48 123 456 789",
+  }],
+});
```

Comment:
Tbh it looks like developer's hell to me and I wouldn't push myself into it, but it's fine, it's your hell, so I don't bother.

However it looks like you could handle most of the data validation here, unless you really want to do it in `UseCase` which is what the `TODO` you left behind suggests.

Reply:
Ofc I understand that this is an “output” dto so in this case there is nothing to validate

Reply:
> Tbh it looks like developer's hell to me

Is it really much worse than [writing a 1.3k line JSON file by hand](https://github.com/RustyRipper/online-city-tickets/blob/main/online-city-tickets-backend/src/main/resources/static/openapi.json)?

> However it looks like you could handle most of the data validation here

That's what happens, supported by `defaultHook` in `/api/shared/hooks.ts`. Both bodies, path parameters, query parameters and headers will (or actually are) automatically checked in accordance to the schema.

In fact, apart from validating request data at runtime, it validates my code at compile-time. I can't return data non-conforming to the OpenAPI schema without a compilation error. If I were to change `getAccountByIdHandler` in `/api/int/routes/get-account-by-id-route.ts` to return `201` instead of `200`, the following thing happens:
![obraz](https://github.com/user-attachments/assets/0fb6d8bf-5bc0-4b6b-a8a7-0d66a81bbacf)

Reply:
@jakubzehner totally forgot about the second part of the question:

> unless you really want to do it in UseCase which is what the TODO you left behind suggests.

This place only support dependency-less, synchronous validation. So I can check whether an email has valid format here, but not whether it is unique (since it requires a call to the DB). So I might end up doing format validation here and the rest in use cases.